subsys/usb/usb_device.c

```diff
@@ -972,6 +972,73 @@ static bool is_ep_valid(uint8_t ep)
 	return false;
 }
 
+static bool usb_get_status_endpoint(struct usb_setup_packet *setup,
+				    int32_t *len, uint8_t **data_buf)
+{
+	uint8_t ep = setup->wIndex;
+	uint8_t *data = *data_buf;
+
+	/* Check if request addresses valid Endpoint */
+	if (!is_ep_valid(ep)) {
+		return false;
+	}
+
+	/** This request is valid for Control Endpoints when
+	 * the device is not yet configured. For other
+	 * Endpoints the device must be configured.
+	 * Firstly check if addressed ep is Control Endpoint.
+	 * If no then the device must be in Configured state
+	 * to accept the request.
+	 */
+	if ((USB_EP_GET_IDX(ep) == 0) || is_device_configured()) {
+		/* bit 0 - Endpoint halted or not */
+		usb_dc_ep_is_stalled(ep, &data[0]);
+		data[1] = 0U;
+		*len = 2;
+		return true;
+	}
+
+	return false;
+}
+
+
+static bool usb_halt_endpoint(struct usb_setup_packet *setup, bool halt)
```

Comment:
this prototype looks like its halting the endpoint however its just EP_HALT request.
Maybe it would be better to name it `usb_ep_halt_req`?